queries/dart/highlights.scm

```diff
@@ -0,0 +1,104 @@
+(dotted_identifier_list) @string
+
+; Methods
+
+(function_type
+  name: (identifier) @method)
+(super) @function
+
+; Annotations
+; TODO: highlight??
+(annotation
+  name: (identifier) @attribute)
+(marker_annotation
+  name: (identifier) @attribute)
+
+; TODO: operators
+[
+ "@" 
+ (additive_operator)
+] @operator
+; TODO: delimiers/punctuation
+
+; Types
+
+(class_definition
+  name: (identifier) @type)
+(constructor_signature
+  name: (identifier) @type)
+(type_identifier
+  (identifier) @type)
+(scoped_identifier
+  scope: (identifier) @type)
+(function_signature
+  name: (identifier) @method)
+(enum_declaration
+  name: (identifier) @type)
+(enum_constant
+  name: (identifier) @type)
+(type_identifier) @type
+(void_type) @type
+
+((scoped_identifier
+  scope: (identifier) @type
+  name: (identifier) @type)
+ (#match? @type "^[a-zA-Z]"))
+
+(type_identifier) @type
+
+; Variables
+
+((identifier) @constant
+ (#match? @constant "^_*[A-Z][A-Z\d_]+"))
+
+(this) @constant.builtin
+
+;; Parameters
+
+(formal_parameter
+    name: (identifier) @parameter)
+
+;; Literals
+
+(hex_integer_literal) @number
+(decimal_integer_literal) @number
+(decimal_floating_point_literal) @float
+;; Those node do not seem to exist
+;(octal_integer_literal) @number
+;(hex_floating_point_literal) @float
```

Comment:
Then we can remove this completely

Reply:
But they should exist as something different... If it's possible to write octals they should appear as something different (`08` at least is parsed a decimal)

Reply:
confirmed. they do not exist in `grammar.js`.